server/selective_sync_test.go

```diff
@@ -0,0 +1,444 @@
+//go:build exclude
```

Comment:
See https://mattermost.atlassian.net/browse/MM-57018 for the remaining work here, but these tests pass locally on my machine.